### PR TITLE
{compiler}[GCCcore/10.2.0,GCCcore/10.3.0] AOCC v3.1.0

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by robert-mijakovic** [comment]
@boegelbot please test @ generoso

**Comment by robert-mijakovic** [comment]
Test report by @robert-mijakovic
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
login01.meluxina.lxp.lu - Linux RHEL 8.3, x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/6cbfa5a4a5b432d1c77a5f97a2a340ba for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/469f24180945f9bc3690f51d3604ace3 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/145c158c579084e7f0c75b4bda157173 for a full test report.

File: easybuild/easyconfigs/a/AOCC/AOCC-3.1.0-GCCcore-10.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+version = '3.1.0'
+
+homepage = 'https://developer.amd.com/amd-aocc/'
+description = "AMD Optimized C/C++ & Fortran compilers (AOCC) based on LLVM 12.0"
+
+# Clang also depends on libstdc++ during runtime, but this dependency is
+# already specified as the toolchain.
+toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
+
+source_urls = ['http://developer.amd.com/wordpress/media/files/']
```
Comment (verdurin) [inline easybuild/easyconfigs/a/AOCC/AOCC-3.1.0-GCCcore-10.2.0.eb:14]: @robert-mijakovic maybe worth adding a comment here about the EULA, to be consistent with other similar easyconfigs?

Comment (robert-mijakovic) [reply easybuild/easyconfigs/a/AOCC/AOCC-3.1.0-GCCcore-10.2.0.eb:16]: Done

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @robert-mijakovic !

**Pull request merged**